From 7a2d0d63c3b6e0e394ab70dae9d6fb70eceb544d Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Sun, 28 Jun 2009 21:31:46 +0200
Subject: [PATCH] Simplify udev/hal (des)activation

	* Only check for HAL if we actually want to build its support.
	* Really on autoconf defined variables/behavior.
---
 configure.ac |   56 ++++++++++++++++++++------------------------------------
 1 files changed, 20 insertions(+), 36 deletions(-)

diff --git a/configure.ac b/configure.ac
index 2eb7e86..dbd560f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -23,35 +23,29 @@ PKG_CHECK_MODULES(LOAD, [libusb])
 AC_SUBST(LOAD_CFLAGS)
 AC_SUBST(LOAD_LIBS)
 
-PKG_CHECK_MODULES(HAL, [hal])
-AC_SUBST(HAL_CFLAGS)
-AC_SUBST(HAL_LIBS)
-
 dnl whether to provide udev rules and loader
 AC_ARG_ENABLE(udev,
-	     [AC_HELP_STRING([--disable-udev],[Disable udev firmware loader])],
-	     [if test $enableval = "yes" ; then
-	     	 SUPPORT_UDEV=1
-	      else
-		SUPPORT_UDEV=0
-	      fi],
-	      [SUPPORT_UDEV=1])
-AM_CONDITIONAL(SUPPORT_UDEV, [test ${SUPPORT_UDEV} = 1])
-if test ${SUPPORT_UDEV} = 1 ; then
+	AS_HELP_STRING([--disable-udev],[Disable udev firmware loader])],
+	, [enable_udev=yes])
+
+AM_CONDITIONAL(SUPPORT_UDEV, [test $enable_udev = yes])
+if test $enable_udev = yes ; then
    udevdir=${libdir}/udev
    AC_SUBST(udevdir)
 fi
 
 dnl whether to provide hal callout
 AC_ARG_ENABLE(hal,
-	     [AC_HELP_STRING([--enable-hal],[Enable HAL callout])],
-	     [if test $enableval = "yes" ; then
-	     	 SUPPORT_HAL=1
-	      else
-		SUPPORT_HAL=0
-	      fi],
-	      [SUPPORT_HAL=0])
-AM_CONDITIONAL(SUPPORT_HAL, [test ${SUPPORT_HAL} = 1])
+	[AS_HELP_STRING([--enable-hal],[Enable HAL callout])],
+	, [enable_hal=no])
+
+if test $enable_hal = yes; then
+	PKG_CHECK_MODULES(HAL, [hal])
+	AC_SUBST(HAL_CFLAGS)
+	AC_SUBST(HAL_LIBS)
+fi
+
+AM_CONDITIONAL(SUPPORT_HAL, [test $enable_hal = yes])
 
 dnl GETTEXT
 GETTEXT_PACKAGE=AC_PACKAGE_NAME
@@ -64,24 +58,14 @@ AM_GLIB_GNU_GETTEXT
 
 IT_PROG_INTLTOOL([0.35])
 
-
-dnl OUTPUT
 AC_OUTPUT
 
-
-yes_no () {
-    if test $1 = 0 ; then
-	echo "no";
-    else
-	echo "yes"
-    fi
-}
-
 echo
-echo "HAL callout :		" $(yes_no $SUPPORT_HAL)
-echo "udev rules :		" $(yes_no $SUPPORT_UDEV)
+echo "HAL callout :		" $enable_hal
+echo "udev rules :		" $enable_udev
 echo
-if test $SUPPORT_UDEV = 1 -a $SUPPORT_HAL = 1 ; then
+
+if test $enable_udev = yes -a $enable_hal = yes ; then
    echo `tput smso` "/!\ WARNING: avoid installing both UDEV rules and HAL callout!" `tput rmso`;
    echo
-fi
\ No newline at end of file
+fi
-- 
1.6.3.3

